src/layers/microscopy-viewer-layer/data-utils/tiff-utils.js

```diff
@@ -1,50 +1,64 @@
+/* eslint-disable no-await-in-loop */
 // eslint-disable-next-line import/extensions
 import { fromUrl } from 'geotiff/dist/geotiff.bundle.min.js';
 
-async function loadTile({ image, channel, x, y, pool }) {
+async function getChannelConnections(channelUrl) {
+  const tiff = await fromUrl(channelUrl);
+  // Get the first image and check its size.
+  const pyramid = await tiff.parseFileDirectories();
+  const maxLevel = pyramid.length;
+  const pyramidLevels = [];
+  for (let i = 0; i < maxLevel; i += 1) {
```

Comment:
Adding a python-ish `range` in utils would let you do this with map.

Reply:
Good idea.